js/data.js

```diff
@@ -1,62 +1,177 @@
-let bookList = [];
+/* eslint max-classes-per-file: ["error", 2] */
 
-const book = {};
+class Book {
+  constructor(title, author, next = null) {
+    this.title = title;
+    this.author = author;
+    this.next = next;
+  }
+}
+
+class BookList {
+  constructor() {
+    this.head = null;
+  }
+
+  prepend(title, author) {
+    const newhead = new Book(title, author);
+    newhead.next = this.head;
+    this.head = newhead;
+  }
+
+  add(title, author) {
+    const book = new Book(title, author);
+    if (this.head === null) {
+      this.head = book;
+      return;
+    }
+
+    let current = this.head;
+
+    while (current.next !== null) {
+      current = current.next;
+    }
+    current.next = book;
+  }
+
+  remove(title, author) {
+    if (this.head === null) { return; }
+    let current = this.head;
+    let previous = null;
+
+    while (current !== null) {
+      if (current.title === title && current.author === author) {
+        if (previous === null) {
+          this.head = current.next;
+        } else {
+          previous.next = current.next;
+        }
+      }
+      previous = current;
+      current = current.next;
+    }
+  }
```

Comment:
- [ ] When I'm removing a book from the list, I get undefined for the books data and the book doesn't get removed. Please fix this issue.

![image](https://user-images.githubusercontent.com/83803414/204631672-f8cdad2b-85f5-461b-93d5-fe52e96cfe87.png)